feat(Condensed): the presheaf of locally constant maps as a condensed set

dagurtomas opened the pull request:
We define the functor `Condensed.LocallyConstant.functor : Type (u+1) ⥤ CondensedSet.{u}` which sends a set `X` to the sheaf of locally constant maps into `X`. 

Further, we prove that this functor is left adjoint to `Condensed.underlying`, and hence isomorphic to
`Condensed.discrete`. The unit of this adjunction is an isomorphism, which yields a proof that
`Condensed.discrete` is fully faithful.

---
- [x] depends on: #12155 
- [x] depends on: #12182 
- [x] depends on: #12186 
- [ ] depends on: #12870 

<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

dagurtomas committed:
chore(Topology): make LocallyConstant.comap computable

dagurtomas committed:
feat(Condensed): the presheaf of locally constant maps as a condensed set

dagurtomas committed:
lint

dagurtomas committed:
lint

dagurtomas committed:
Merge branch 'master' into dagur_CondensedLC

grunweg committed:
chore(GroupTheory): remove autoImplicit (#12134)

dagurtomas committed:
Merge branch 'master' into dagur_CondensedLC

urkud committed:
chore(Order/JordanHolder): `List.nthLe` → `List.get` (#12208)

urkud committed:
chore(PeriodicPts): migrate `List.nthLe` → `List.get` (#12209)

xroblot committed:
feat: Define the covolume of a `ZLattice` (#11327)

This PR defines the covolume of a `Zlattice` as the (finite) volume of any of its fundamental domain and proves some results about it, mainly: it's nonzero, it does not depend of the choice of the fundamental domain and it is equal to the absolute value of the determinant of any basis of the lattice. 

This PR also creates a directory `Zlattice` (with the original file becoming `Zlattice/Basic.lean`) and a new file `Zlattice/Covolume.lean`.





Co-authored-by: Xavier Roblot <roblot@Algorithmi.local>
Co-authored-by: Xavier Roblot <roblot@math.univ-lyon1.fr>

eric-wieser committed:
chore(Util/Superscript): delete deleteme section (#12212)

leanprover/lean4#2269 was shipped long ago, so this isn't needed.

dagurtomas committed:
feat(CategoryTheory): a lemma about composing different isomorphisms from opposite coproducts to products (#12155)

joelriou commented on Mathlib/Condensed/Discrete/LocallyConstant.lean:
```diff
@@ -0,0 +1,499 @@
+-/
+noncomputable def functorIsoTopCatToCondensed : functor ≅ TopCat.discrete ⋙ topCatToCondensed :=
+  NatIso.ofComponents (fun X ↦ {
+    hom := ⟨(functorToPresheavesIsoTopCatToCondensed X).hom⟩
```
It should be possible to use `Functor.preimageIso` here (applied to the forget functor from sheaves to presheaves).

dagurtomas replied:
That works, thanks!

judithludwig committed:
feat(RingTheory/Etale): add definition of unramified, smooth and etale (#12044)

Adds definitions for unramified, smooth and étale algebras and shows basic permanence properties.

In a follow-up PR we show that these notions are stable under base change.

Co-authored-by: Judith Ludwig <ludwig@mathi.uni-heidelberg.de>



Co-authored-by: Christian Merten <christian@merten.dev>
Co-authored-by: judithludwig <150042763+judithludwig@users.noreply.github.com>

dagurtomas committed:
Merge branch 'master' into dagur_CondensedLC

dagurtomas committed:
apply suggestion from Joel

dagurtomas commented:
Marked as WIP because I want to generalise a bit so that it more easily applies to light condensed sets as well

dagurtomas committed:
feat(Condensed): prove the explicit sheaf condition for a completely general target category

dagurtomas committed:
fix typo

dagurtomas committed:
cleanup

dagurtomas committed:
forgetful stuff

dagurtomas committed:
move reflects classes to correct file

dagurtomas committed:
modify whole preserves finite file

dagurtomas committed:
remove commented defs

dagurtomas committed:
add back stuff

dagurtomas committed:
Update Explicit.lean

dagurtomas committed:
Merge branch 'dagur_CondensedComp' into dagur_CondensedLC

dagurtomas committed:
generalise universes

dagurtomas committed:
.

dagurtomas commented:
Incorporated into #14027
